Automattic/wp-calypso
# Add routine to claim discount during cancellation

#### Proposed Changes

* This PR builds on the offer step PR to add the logic to claim the discount presented.

#### Testing Instructions



* If you have tested this functionality before, you may need to follow some additional steps first: pbNhbs-2Ep-p2#comment-9629
* You will need a Jetpack site to test with and a Jetpack product/ plan purchase. Try testing with a new purchase.
* Navigate to `/me/purchases` to view the list of your purchases, select your purchase from the list.
* From the individual purchase page, click to cancel/ remove your purchase to start the cancellation flow. Example:
![Screen Shot 2022-07-28 at 9 34 41 AM](https://user-images.githubusercontent.com/18016357/181520644-d00dbcde-0e58-4268-be87-d861b2541ec2.png)

* Follow the steps in the cancellation flow, but do not remove the purchase. 
* You should see a screen similar to this one that presents a special offer:
![Screen Shot 2022-07-07 at 12 22 28 PM](https://user-images.githubusercontent.com/18016357/177872515-2eb89fce-4f49-4831-9414-178ef5cbdc0a.png)

* Make sure all the links provided in the text work as expected and that the copy makes sense.
* Take note of the discounted rate, as you'll need to compare the rate presented here to the rate displayed on your purchase after claiming the offer.
* From this screen, click on the "Get Discount" button
* The "Get Discount" button should switch to show "Getting Discount" and display a waiting/ loading animation
* If successful, you should see a confirmation screen that looks like this:
![Screen Shot 2022-07-26 at 12 05 12 PM](https://user-images.githubusercontent.com/18016357/181520580-26be7f09-42ab-40a8-8dbc-1f0ecdf4b74d.png)

* Clicking on "Back to my website" or clicking out of the modal should return you to the `/me/purchases` page.
* Locate the product purchase for your test site and click on it to view the individual purchase page.
* Confirm that the new discounted rate shows on the purchase page and matches the rate shown when the discount was claimed. You should also see information related to the offer in the "Price" section of the page:
![Screen Shot 2022-07-28 at 9 42 52 AM](https://user-images.githubusercontent.com/18016357/181520860-defd4794-5a69-4cdd-a48d-c5ae3f8de90b.png)


#### Pre-merge Checklist
